src/gatsby/src/components/search/cityNameSearch.js

```diff
@@ -0,0 +1,221 @@
+import React, { useState, useEffect, useRef } from "react"
+import { graphql, useStaticQuery } from "gatsby"
+import { Typeahead, ClearButton, Menu, MenuItem } from "react-bootstrap-typeahead"
+import { Form } from "react-bootstrap"
+import PermissionToast from "./permissionToast"
+import NearMeIcon from "@mui/icons-material/NearMe"
+import "react-bootstrap-typeahead/css/Typeahead.css"
+
+const HighlightText = ({ city, input }) => {
+  const words = city.split(" ")
+  return (
+    words.map((word, index) => {
+      if (word.toLowerCase() === input) {
+        return <span key={index}> {word} </span>
+      } else {
+        return <b key={index}> {word} </b>
+      }
+    })
+  )
+}
+
+const CityNameSearch = ({
+  showPosition, currentLocation, optionLimit, selectedItems, handleChange, handleKeyDownSearch, handleClick, handleKeyDown
+}) => {
+  const data = useStaticQuery(graphql`
+    query {
+      allStrapiSearchCity(
+        sort: {rank: ASC},
+        filter: {rank: {lte: 4}}
+      ) {
+        nodes {
+          strapi_id
+          cityName
+          latitude
+          longitude
+          rank
+        }
+      }
+    }
+  `)
+
+  // useState and constants
+  const [cityText, setCityText] = useState("")
+  const [hasResult, setHasResult] = useState(false)
+  const [hasPermission, setHasPermission] = useState(true)
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
+  const cities = data?.allStrapiSearchCity?.nodes || []
+  const typeaheadRef = useRef(null)
+
+  // functions
+  const cityOptions = (optionLimit) => {
+    const cityTextLower = cityText.toLowerCase()
+    const filteredCities = cities.filter(
+      city => city.cityName.toLowerCase().includes(cityTextLower)
```

Comment:
Let's try changing the matching criteria here so searching for "lan" doesn't bring back "island"
```
      city => city.cityName.toLowerCase().startsWith(cityTextLower) || city.cityName.toLowerCase().includes(` ${cityTextLower}
```